backend/models/office_hours/oh_event.py

```diff
@@ -0,0 +1,27 @@
+from pydantic import BaseModel
+from datetime import datetime, date
+from oh_type import OfficeHoursType
+
+__authors__ = ["Sadie Amato"]
+__copyright__ = "Copyright 2024"
+__license__ = "MIT"
+
+
+class OfficeHoursEvent(BaseModel):
+    """
+    Pydantic model to represent an `OfficeHoursEvent` that has not been created yet.
+
+    This model is based on the `OfficeHoursEventEntity` model, which defines the shape
+    of the `OfficeHoursEvent` database in the PostgreSQL database.
+    """
+    id: int | None = None
```

Comment:
I might suggest making this slightly more nuanced so that you can avoid null checks later on. I would recommend:

`OfficeHoursDraft(BaseModel)` - This does not have an `id` field but has everything else you would want to see in a draft version of an event (before it is persisted in the backend).

`OfficeHoursEvent(OfficeHoursDraft)` - This type adds the id field (and any other fields that would only exist in a persisted version, if there are differences).

Reply:
Good point, thank you! With the naming, would you prefer us to use ObjDraft and Obj over using Obj and OBjDetails (where the second name is the one adding on fields)? I saw both used throughout the codebase and was wondering if there were actual use case differences

Reply:
Annoyingly: yes :)

There's actually a sound argument for having three models for the same entity depending on usage:

1. Draft - pre-persistence
2. Plain - Just the model persisted, minimal/no relationships loaded
3. Details - Full model with relationships

The plain models tend to be what's loaded in relationships of a details model (to avoid circularity in details models). For example, ticket details may load info on users via plain models but not user details models (which may load related plain tickets but not details tickets)

Reply:
This has been updated now in our recent commit, thank you!!